### ISSUE TITLE
FailOnError for legends

### ISSUE THREAD

**Issue by pvalsecc** [mapfish/mapfish-print#411]
Unlike the map tiles, when legends are failing, we cannot make the print fail.

**Comment by margitbomholt** [comment]
We haven't been able to get the print fail, on errors, eg. wrong layer name/or service not responding, even though we have added a  boolean property "failOnError" to the layer in the son request. 
Do you know whether that should be possible?

**Comment by pvalsecc** [comment]
Yes, you are right @margitbomholt. From what I see in the code, `failOnError` is used only for tiled layers. So yes, we should include non-tiled and vector layers.

**Comment by margitbomholt** [comment]
Thank you for answering. In our test we have used tiled WMS layers, but we haven't managed to get any error messages.. Is there any settings that has to be done on the mapfish server,? A snip of our request  json : {"baseURL":"http://kortforsyningen.kms.dk/service","customParams":{"login":"xxx","password":"xxx","servicename":"topo_fot","VERSION":"1.1.1","TRANSPARENT":"true","LAYERS":"basis","SERVICE":"WMS","SRS":"EPSG:25832","buffer":50,"TILED":"true"},"layers":["basis"],"opacity":1,"styles":[""],"type":"WMS","failOnError":true}.

**Comment by pvalsecc** [comment]
As far as I know, there is no additional setting needed on the server side.

**Comment by pvalsecc** [comment]
#834 improved a bit the situation, here.

**Comment by pierrejego** [comment]
Hello,
Thanks for this work, I have made an additional modification to manage exchange with QGisServer.
When Layer name does not exist on qgiserver, QgisServer will respond with http 200 btu with some XML.
 
I have made a PR #996, review is welcome :)
